From e56e92683a7ff0ff87dd29c8538e02e63bc7c79e Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@freescale.com>
Date: Fri, 27 Feb 2015 16:22:46 +0800
Subject: [PATCH] MLK-10258-2 IMX7D: EPXP: add make support for the new PXP
 module on 7D board

Modify some related Kconfig and Makefiles to make the
EPXP driver can be build and also allows the PxP V2
driver can co-exsit with EPXP driver.

Signed-off-by: Fancy Fang <chen.fang@freescale.com>
---
 drivers/dma/Kconfig      | 11 ++++++++++-
 drivers/dma/Makefile     |  1 +
 drivers/dma/pxp/Makefile |  1 +
 3 files changed, 12 insertions(+), 1 deletion(-)

diff --git a/drivers/dma/Kconfig b/drivers/dma/Kconfig
index 88a4988..acdeecd 100644
--- a/drivers/dma/Kconfig
+++ b/drivers/dma/Kconfig
@@ -145,10 +145,19 @@ config MXC_PXP_V2
           Support the PxP (Pixel Pipeline) on i.MX6 DualLite and i.MX6 SoloLite.
 	  If unsure, select N.
 
+config MXC_PXP_V3
+	bool "MXC PxP V3 support"
+	depends on ARM
+	select DMA_ENGINE
+	help
+	  Support the PxP V3(Pixel Pipeline) on i.MX7D. The PxP V3 supports
+	  more functions than PxP V2, dithering, reagl/-D and etc.
+	  If unsure, select N.
+
 config MXC_PXP_CLIENT_DEVICE
 	bool "MXC PxP Client Device"
 	default y
-	depends on MXC_PXP_V2
+	depends on MXC_PXP_V2 || MXC_PXP_V3
 
 config TXX9_DMAC
 	tristate "Toshiba TXx9 SoC DMA support"
diff --git a/drivers/dma/Makefile b/drivers/dma/Makefile
index 0c5ef31..0bd17ef 100644
--- a/drivers/dma/Makefile
+++ b/drivers/dma/Makefile
@@ -19,6 +19,7 @@ obj-$(CONFIG_DW_DMAC_CORE) += dw/
 obj-$(CONFIG_AT_HDMAC) += at_hdmac.o
 obj-$(CONFIG_MX3_IPU) += ipu/
 obj-$(CONFIG_MXC_PXP_V2) += pxp/
+obj-$(CONFIG_MXC_PXP_V3) += pxp/
 obj-$(CONFIG_TXX9_DMAC) += txx9dmac.o
 obj-$(CONFIG_SH_DMAE_BASE) += sh/
 obj-$(CONFIG_COH901318) += coh901318.o coh901318_lli.o
diff --git a/drivers/dma/pxp/Makefile b/drivers/dma/pxp/Makefile
index 498ccca..42e4ace 100644
--- a/drivers/dma/pxp/Makefile
+++ b/drivers/dma/pxp/Makefile
@@ -1,2 +1,3 @@
 obj-$(CONFIG_MXC_PXP_V2) += pxp_dma_v2.o
+obj-$(CONFIG_MXC_PXP_V3) += pxp_dma_v3.o
 obj-$(CONFIG_MXC_PXP_CLIENT_DEVICE) += pxp_device.o
-- 
1.8.0

